config/src/main/java/com/typesafe/config/impl/BadMap.java

```diff
@@ -74,7 +74,7 @@ private static <K,V> void store(Entry[] entries, int hash, K k, V v) {
     }
 
     private static void store(Entry[] entries, Entry e) {
-        int i = e.hash % entries.length;
```

Comment:
I think e.hash here is already abs'd (see line 65)...

Reply:
I hit an exception when trying it out in the test.